BUG: stats: XSLOW test failure in `scipy.stats.tests.test_fit::TestFit`

### Describe your issue.

With scipy `'1.15.0.dev0+git20241125.7917286'`, I get a new XSLOW test failure in `stats`.

```
$ SCIPY_XSLOW=1 python3 dev.py test -m full -j 12 -t scipy.stats.tests.test_fit::TestFit
💻  ninja -C /home/warren/repos/git/forks/scipy/build -j12
ninja: Entering directory `/home/warren/repos/git/forks/scipy/build'
[2/338] Generating subprojects/highs/src/HConfig.h with a custom command
Build OK
💻  meson install -C build --only-changed --tags runtime,python-runtime,tests,devel
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages
Running tests for scipy version:1.15.0.dev0+git20241125.7917286, installed at:/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy
==================================== test session starts =====================================
platform linux -- Python 3.12.4, pytest-8.2.2, pluggy-1.5.0
rootdir: /home/warren/repos/git/forks/scipy
configfile: pytest.ini
plugins: hypothesis-6.105.1, timeout-2.3.1, xdist-3.6.1, cov-5.0.0, anyio-4.4.0
12 workers [282 items]    
......s.........s........sss....................s....s..................s............. [ 30%]
s....s..s.....................ss..ss..........................s....s.....s.......s..s. [ 60%]
........s........s...............s..............s..s....sF.s...s...................... [ 91%]
.s......................                                                               [100%]
========================================== FAILURES ==========================================
_____________________________ TestFit.test_basic_fit_mse[rdist] ______________________________
[gw1] linux -- Python 3.12.4 /home/warren/py3.12.4/bin/python3
scipy/stats/tests/test_fit.py:539: in test_basic_fit_mse
    self.basic_fit_test(dist_name, "mse", rng=2)
        dist_name  = 'rdist'
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x76ac117fcc50>
scipy/stats/tests/test_fit.py:530: in basic_fit_test
    assert_nlff_less_or_close(dist, data, res.params, ref, **self.tols,
        N          = 5000
        bounds     = array([[1.6e-01, 1.6e+01],
       [0.0e+00, 1.0e+01],
       [1.0e-16, 1.0e+01]])
        data       = array([5.57563493, 5.11340866, 4.07652048, ..., 4.72223674, 5.54069523,
       5.08541462])
        dist       = <scipy.stats._continuous_distns.rdist_gen object at 0x76ac437334d0>
        dist_data  = {'alpha': (3.570477051665046,), 'anglit': (), 'arcsine': (), 'argus': (1.0,), ...}
        dist_name  = 'rdist'
        loc        = 4.790547883456026
        method     = 'mse'
        nlff_name  = '_penalized_nlpsf'
        nlff_names = {'mle': 'nnlf', 'mse': '_penalized_nlpsf'}
        ref        = [1.6, 4.790547883456026, 0.9992905124308905]
        res        =   params: FitParams(c=np.float64(5.58494761552835), loc=np.float64(4.718162428437992), scale=np.float64(1.59208361666767))
 success: True
 message: 'Optimization terminated successfully.'
        rng        = Generator(PCG64) at 0x76AC117002E0
        scale      = 0.9992905124308905
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x76ac117fcc50>
        shapes     = array([1.6])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x76ac1174ffe0>
scipy/stats/tests/test_fit.py:366: in assert_nlff_less_or_close
    np.testing.assert_allclose(nlff1, nlff0, rtol=rtol, atol=atol)
        atol       = 0.05
        data       = array([5.57563493, 5.11340866, 4.07652048, ..., 4.72223674, 5.54069523,
       5.08541462])
        dist       = <scipy.stats._continuous_distns.rdist_gen object at 0x76ac437334d0>
        nlff       = <bound method rv_generic._penalized_nlpsf of <scipy.stats._continuous_distns.rdist_gen object at 0x76ac437334d0>>
        nlff0      = np.float64(45544.1473094238)
        nlff1      = np.float64(46612.83874403211)
        nlff_name  = '_penalized_nlpsf'
        params0    = [1.6, 4.790547883456026, 0.9992905124308905]
        params1    = FitParams(c=np.float64(5.58494761552835), loc=np.float64(4.718162428437992), scale=np.float64(1.59208361666767))
        rtol       = 0.01
/home/warren/py3.12.4/lib/python3.12/contextlib.py:81: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=0.01, atol=0.05
E   
E   Mismatched elements: 1 / 1 (100%)
E   Max absolute difference among violations: 1068.69143461
E   Max relative difference among violations: 0.02346496
E    ACTUAL: array(46612.838744)
E    DESIRED: array(45544.147309)
        args       = (<function assert_allclose.<locals>.compare at 0x76ac117282c0>, array(46612.83874403), array(45544.14730942))
        func       = <function assert_array_compare at 0x76ac53fb7c40>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0.01, atol=0.05', 'strict': False, ...}
        self       = <contextlib._GeneratorContextManager object at 0x76ac53fd26c0>
================================== short test summary info ===================================
FAILED scipy/stats/tests/test_fit.py::TestFit::test_basic_fit_mse[rdist] - AssertionError: 
========================= 1 failed, 252 passed, 29 skipped in 58.94s =========================

```


### Reproducing Code Example

```python
N/A
```


### Error message

```shell
See above.
```


### SciPy/NumPy/Python version and system information

```shell
In [1]: import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info
   ...: ); scipy.show_config()

1.15.0.dev0+git20241125.7917286 2.1.3 sys.version_info(major=3, minor=12, micro=4, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.13.2
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  pythran:
    include directory: ../../../../../py3.12.4/lib/python3.12/site-packages/pythran
    version: 0.16.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/warren/py3.12.4/bin/python3
  version: '3.12'
```

--- a/scipy/stats/tests/test_fit.py
+++ b/scipy/stats/tests/test_fit.py
@@ -284,7 +284,7 @@ def cases_test_fit_mse():
                       'gausshyper', 'genhyperbolic',  # integration warnings
                       'tukeylambda',  # close, but doesn't meet tolerance
                       'vonmises',  # can have negative CDF; doesn't play nice
-                      'arcsine', 'argus', 'powerlaw',  # don't meet tolerance
+                      'arcsine', 'argus', 'powerlaw', 'rdist', # don't meet tolerance
                       'poisson_binom',  # vector-valued shape parameter
                       }
 
@@ -314,7 +314,7 @@ def cases_test_fit_mse():
                        'nakagami', 'ncf', 'nchypergeom_fisher',
                        'nchypergeom_wallenius', 'nct', 'ncx2',
                        'pearson3', 'powerlognorm',
-                       'rdist', 'reciprocal', 'rel_breitwigner', 'rice',
+                       'reciprocal', 'rel_breitwigner', 'rice',
                        'trapezoid', 'truncnorm', 'truncweibull_min',
                        'vonmises_line', 'zipfian'}
 

